src/backend/drm/gbm/mod.rs

```diff
@@ -74,8 +73,9 @@ static LOAD: Once = Once::new();
 
 /// Representation of an open gbm device to create rendering surfaces
 pub struct GbmDevice<D: RawDevice + ControlDevice + 'static> {
-    pub(self) dev: Rc<RefCell<gbm::Device<D>>>,
-    backends: Rc<RefCell<HashMap<crtc::Handle, Weak<GbmSurfaceInternal<D>>>>>,
+    pub(self) dev: Arc<Mutex<gbm::Device<gbm::FdWrapper>>>,
```

Comment:
We do not wrap the underlying device in the `gbm::Device` anymore, because we need to share it with the surface. If we would also wrap the `D`, we would also share the `D`, which would mean the device would need to be `Send`, which it isn't.

Note, that this does not violate our drop order. We can now drop the underlying device, while surfaces are still active, BUT a `GbmSurface` still holds a `Legacy/AtomicDrmSurface`, which holds a `Dev`, which is the struct that actually closes the device file descriptor on *Drop* and resets the device. So nothing to worry about here.